index.d.ts

```diff
@@ -0,0 +1,19 @@
+type Points = number[][]
+
+declare class KDBush<T> {
+  ids: number[]
+  coords: number[]
+  nodeSize: number
+  points: T[]
+  range(minX: number, minY: number, maxX: number, maxY: number)
+  within(x: number, y: number, r: number)
+}
+
+interface KDBush<T> {
+  (points: Points): KDBush<Points>
+  <T>(points: T[], getX: (p: T) => void, getY: (p: T) => void, nodeSize?: number, ArrayType?: any): KDBush<T>
+}
+
+declare const kdbush: KDBush<Points>
```

Comment:
Here's an example of the benefits of having type checking with `kdbush`:

## Valid
```typescript
const xy = [{x: 110, y: 60}, {x: 130, y: 40}]
const index = kdbush(xy, p => p.x, p => p.y)
index.points[0].x
index.points[0].y
```
## Not Valid
```typescript
const xy = [{x: 110, y: 60}, {x: 130, y: 40}]
const index = kdbush(xy, p => p.x, p => p.y)
index.points[0].foo
index.points[0].bar
index.points.map(p => [p.foo, p.bar])
```
**errors:**
```
$ tsc types.ts
types.ts(24,17): error TS2339: Property 'foo' does not exist on type '{ x: number;
 y: number; }'.
types.ts(25,17): error TS2339: Property 'bar' does not exist on type '{ x: number;
 y: number; }'.
types.ts(26,26): error TS2339: Property 'foo' does not exist on type '{ x: number;
 y: number; }'.
types.ts(26,33): error TS2339: Property 'bar' does not exist on type '{ x: number;
 y: number; }'.
```